src/promptflow/promptflow/_sdk/entities/_connection.py

```diff
@@ -618,6 +625,113 @@ def _get_schema_cls(cls):
         return FormRecognizerConnectionSchema
 
 
+class CustomStrongTypeConnection(_Connection):
+    """Custom strong type connection.
+
+    :param configs: The configs kv pairs.
+    :type configs: Dict[str, str]
+    :param secrets: The secrets kv pairs.
+    :type secrets: Dict[str, str]
+    :param name: Connection name
+    :type name: str
+
+    This connection type should not be used directly. Below is an example of how to use CustomStrongTypeConnection:
+    class MyCustomConnection(CustomStrongTypeConnection):
+        api_key: Secret
+        api_base: str
+    """
+
+    def __init__(
+        self,
+        secrets: Dict[str, str],
+        configs: Dict[str, str] = None,
+        **kwargs,
+    ):
+        # There are two cases to init a Custom strong type connection:
+        # 1. The connection is created through SDK PFClient, custom_type and custom_module are not in the kwargs.
+        # 2. The connection is loaded from template file, custom_type and custom_module are in the kwargs.
+        custom_type = kwargs.get(CustomStrongTypeConnectionConfigs.TYPE, None)
+        custom_module = kwargs.get(CustomStrongTypeConnectionConfigs.MODULE, None)
+        if custom_type:
+            configs.update({CustomStrongTypeConnectionConfigs.PROMPTFLOW_TYPE_KEY: custom_type})
+        if custom_module:
+            configs.update({CustomStrongTypeConnectionConfigs.PROMPTFLOW_MODULE_KEY: custom_module})
+        self.kwargs = kwargs
+        super().__init__(configs=configs, secrets=secrets, **kwargs)
+        self.module = kwargs.get("module", self.__class__.__module__)
+        self.custom_type = custom_type or self.__class__.__name__
+
+    def is_custom_strong_type(self):
+        return True
+
+    def _to_orm_object(self) -> ORMConnection:
+        custom_connection = self.convert_to_custom()
+        return custom_connection._to_orm_object()
+
+    def convert_to_custom(self):
```

Comment:
are we intent to make this public? add docstring for public apis

Reply:
make it private if we don't intent customer to use it

Reply:
add _ to make it private